modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java

```diff
@@ -635,6 +635,66 @@ public RpcMsg eth_sign(Object _params) {
         return new RpcMsg(TypeConverter.toJsonHex(key.sign(messageHash).getSignature()));
     }
 
+    /**
+     * Sign a transaction. This account needs to be unlocked.
+     * @param _params
+     * @return
+     */
+    public RpcMsg eth_signTransaction(Object _params) {
+        JSONObject _tx;
+        //Address to sign with
+        String _address = null;
+        if (_params instanceof JSONArray) {
+            _tx = ((JSONArray) _params).getJSONObject(0);
+
+            if(((JSONArray) _params).length() > 1)
+                _address = ((JSONArray) _params).get(1) + "";
+        } else if (_params instanceof JSONObject) {
+            _tx = ((JSONObject) _params).getJSONObject("transaction");
+            _address = ((JSONObject) _params).get("address") + "";
+        } else {
+            return new RpcMsg(null, RpcError.INVALID_PARAMS, "Invalid parameters");
+        }
+
+        ArgTxCall txParams = ArgTxCall.fromJSON(_tx, getNrgOracle(), getDefaultNrgLimit());
+        if (txParams == null)
+            return new RpcMsg(
+                null, RpcError.INVALID_PARAMS, "Please check your transaction object.");
+
+        Address address = null;
+        if(_address == null || _address.trim().length() == 0)
+            return new RpcMsg(null, RpcError.INVALID_PARAMS, "Address cannot be null");
+        else
+            address = Address.wrap(_address);
+
+        ECKey key = getAccountKey(address.toString());
```

Comment:
Can you please move things like the nullity checks, and the account unlocked check to the ApiAion.signTransaction() function? 

The reason is that we also support a different API, and this would avoid code duplication. For now, simply log and error and return null from the ApiAion.signTransaction() when necessary. 

Also I'd suggest sending the address as a string to ApiAion, and doing the key extraction there.